orchestrator/packages/api/src/controllers/grading-queue-controller.ts

```diff
@@ -175,3 +176,22 @@ export const deleteJob = async (req: Request, res: Response) => {
     }
   }
 };
+
+export const getJobStatus = async (req: Request, res: Response) => {
+  const { key, response_url } = req.body;
+  const jobQueueStatus = await getJobQueueStatus(key, response_url);
+  if (!jobQueueStatus) {
+    return res.json("We could not find the job you're looking for. Please contact a professor or admin.");
+  }
+  const { reservation, numReservationsAhead } = jobQueueStatus;
+  const now = new Date();
+  if (now > reservation.releaseAt) {
+    const minutesUntilRelease = Math.floor((now.getTime() - reservation.releaseAt.getTime()) / 60_000);
+    return res.json(`Your job will be released in ${minutesUntilRelease ? (minutesUntilRelease.toString() + " minutes.") : "less than a minute."}`);
+  } else if (numReservationsAhead === 0) {
```

Comment:
Replace these last two cases with "You are `#1`, `#2`, `#N` in the queue to be graded."